staging/src/k8s.io/apiserver/pkg/registry/rest/update.go

```diff
@@ -51,6 +52,19 @@ type RESTUpdateStrategy interface {
 	// filled in before the object is persisted.  This method should not mutate
 	// the object.
 	ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList
+	// WarningsOnUpdate returns any warnings that should be returned to the updater.
+	// WarningsOnUpdate is invoked after default fields in the object have been filled in
+	// and after ValidateUpdate has passed, before Canonicalize is called, and before the object is persisted.
+	// This method must not mutate either object.
+	//
+	// Be brief; limit warnings to 120 characters if possible.
+	// Don't include a "Warning:" prefix in the message (that is added by clients on output).
+	// Warnings returned about a specific field should be formatted as "path.to.field: message".
+	//
+	// Use warning messages to describe problems the client making the API request should correct or be aware of.
```

Comment:
What other guidance can we give?  Maybe examples of how you plan to use it (2-3)?